tiledb/sm/query/readers/sparse_global_order_reader.cc

```diff
@@ -1996,9 +1921,196 @@ void SparseGlobalOrderReader<BitmapType>::process_slabs(
     }
   }
 
+  // If any overflow happened after a field with aggregates, we would need to
+  // recompute the aggregate. For now just throw an exception as this will only
+  // happen in very rare cases.
+  if (last_field_to_overflow.has_value()) {
+    for (auto& name : names) {
+      if (name == last_field_to_overflow.value()) {
+        break;
+      }
+
+      if (aggregates_.count(name) != 0) {
+        for (auto& aggregates : aggregates_[name]) {
+          if (aggregates->need_recompute_on_overflow()) {
+            throw SparseGlobalOrderReaderStatusException(
+                "Overflow happened after aggregate was computed, aggregate "
+                "recompute pass is not yet implemented");
```

Comment:
I've still got no idea how recomputing an overflow won't just overflow the second time around.

Reply:
Hopefully my explanations above help?

Reply:
Also unclear to me (with the comment).

Reply:
@ihnorton just making sure, the comment above was in the PR... Not in the code. It was:
"This is a var size overflow. When we run any query, the user gives us a var buffer when they request a variable length field. When we copy the data to the var buffer, we might realize that we can fit less cells than we previously thought. We then need to scale the results back for any fields that were previously copied as all buffers need to return the same number of cells. We made the decision to use, for now, the same structure to save progress for both the copy operation and the aggregate operation. For a sum operation, that means that if we already summed some data then we later run into a var sized overflow, we need to reprocess the sum so that we don't include more cells in the sum than what are included in the regular buffers. That way cells are not counted twice when we continue the INCOMPLETE query. For some aggregates (MIN/MAX for example), this doesn't matter as the aggregation is an absolute and processing cells twice will yield the same result."

Does that help clarifying? I can add more comments to the code if you'd like.